src/makeInfinityAdder.js

```diff
@@ -6,7 +6,24 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let numbers = [];
+
+  const iterate = (figure) => {
+    if (figure !== undefined) {
+      numbers.push(figure);
+
+      return iterate;
+    } else {
+      const allNumbers = [...numbers];
+      const endResult = allNumbers.reduce((sum, number) => sum + number, 0);
+
+      numbers = [];
+
+      return endResult;
+    }
```

Comment:
try to simplify the solution. `reduce ` means each new loop. You know that the parameters will be numbers, so do it with the help of numbers. `const count = 0` ... `count +=  ...`